packages/storage-plugin/src/keys-manager.ts

```diff
@@ -0,0 +1,69 @@
+import {
+  Injectable,
+  Injector,
+  inject,
+  EnvironmentProviders,
+  ENVIRONMENT_INITIALIZER,
+  makeEnvironmentProviders
+} from '@angular/core';
+import {
+  STORAGE_ENGINE,
+  StorageEngine,
+  StorageKey,
+  ɵextractStringKey,
+  NgxsStoragePluginOptions,
+  ɵisKeyWithExplicitEngine,
+  ɵNGXS_STORAGE_PLUGIN_OPTIONS
+} from '@ngxs/storage-plugin/internals';
+
+interface KeyWithEngine {
+  key: string;
+  engine: StorageEngine;
+}
+
+@Injectable({ providedIn: 'root' })
+export class ɵNgxsStoragePluginKeysManager {
+  readonly keysWithEngines: KeyWithEngine[] = [];
```

Comment:
Exposing this breaks encapsulation. We need to discuss alternatives to exposing this property.